nixos/modules/services/mail/postal.nix

```diff
@@ -0,0 +1,530 @@
+{
+  config,
+  lib,
+  options,
+  pkgs,
+  ...
+}:
+with lib;
+let
+  cfg = config.services.postal;
+  opt = options.services.postal;
+
+  yaml = pkgs.formats.yaml { };
+
+  postalConfigFile = yaml.generate "postal-config.yml" cfg.settings;
+
+  wrapper = pkgs.writeShellScriptBin "postal" (
+    lib.concatStrings [
+      (lib.optionalString (cfg.environmentFile != null) ''
+        export $(cat ${cfg.environmentFile} | xargs)
+      '')
+      ''
+        export POSTAL_CONFIG_FILE_PATH=${postalConfigFile}
+        exec ${pkgs.postal}/bin/postal "$@"
+      ''
+    ]
+  );
+
+  postalCli = pkgs.symlinkJoin {
+    name = "postal";
+    paths = [ wrapper ];
+  };
```

Comment:
What is the idea behind this? As is't just this one package we pass this directly to `environment.systemPackages`

Reply:
My mistake :) I rewrote the wrapper and move it to `environment.systemPackages`